asserts/account_key.go

```diff
@@ -115,6 +138,21 @@ func assembleAccountKey(assert assertionBase) (Assertion, error) {
 		return nil, err
 	}
 
+	// XXX: We should require name to be non-empty after backfilling existing assertions.
+	value, ok := assert.headers["name"]
+	if ok {
+		name, ok := value.(string)
+		if !ok {
+			return nil, fmt.Errorf(`"name" header must be a string`)
+		}
+		if len(name) == 0 {
+			return nil, fmt.Errorf(`"name" header should not be empty`)
+		}
```

Comment:
I think these 2 checks can be replaced by checkNotEmptyString. That also performs a mandatory check, but that should be redundant and always pass for now and is what we eventually want.